staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -471,23 +475,33 @@ func (qs *queueSet) timeoutOldRequestsAndRejectOrEnqueueLocked(ctx context.Conte
 // chooseQueueIndexLocked uses shuffle sharding to select a queue index
 // using the given hashValue and the shuffle sharding parameters of the queueSet.
 func (qs *queueSet) chooseQueueIndexLocked(hashValue uint64, descr1, descr2 interface{}) int {
+	var backHand [8]int
+	// the dealer uses the current desired number of queues, which is no larger than the number in `qs.queues`.
+	hand := qs.dealer.DealIntoHand(hashValue, backHand[:])
```

Comment:
Can you describe what's the purpose of this change?

Reply:
done.